src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientTest.java

```diff
@@ -158,8 +157,8 @@ public void failedWithBearerTokenFromConfiguration() throws Exception {
         try {
             message = saslServer.evaluateResponse(message);
         } catch (SaslException e) {
-            assertTrue(e.getCause() instanceof RealmUnavailableException);
-            RealmUnavailableException cause = (RealmUnavailableException) e.getCause();
+            assertTrue(e.getCause() instanceof IllegalArgumentException);
```

Comment:
This part I am still not completely sure about.

The SASL mechanism throwing SaslException is fine, that is what it is supposed to do but this is relying on an Exception coming all the way from a SecurityRealm through the ServerAuthenticationContext to be handled by the mechanism.